src/wallet/wallet.cpp

```diff
@@ -4781,7 +4895,8 @@ bool CWallet::InitLoadWallet(const CChainParams& params, bool clearWitnessCaches
         walletInstance->SetMaxVersion(nMaxVersion);
     }
 
-    if (!walletInstance->HaveHDSeed())
+    // TODO: should this go in `LoadWallet` instead?
```

Comment:
No. We call `LoadWallet` to load the `wallet.dat` from disk (if it exists), and only afterwards can we safely check that the wallet does not have an existing mnemonic phrase, and generate one.